src/nagini_translation/analyzer.py

```diff
@@ -429,6 +429,57 @@ def find_or_create_target_class(self, node: ast.AST) -> PythonClass:
             raise UnsupportedException(node,
                                        'class literal has unsupported format')
 
+    def _check_is_adt_well_formed(self, adt: PythonClass,
+                                  actual_bases: List[ast.AST],
+                                  node: ast.AST, ast: ast.AST):
+        """
+        Chechs if the ADT is syntactically well-formed, throwing an
+        invalid program exception otherwise.
+        """
+        adt.is_adt = True
```

Comment:
This is a small detail, but IMO setting of the is_adt field should not happen in the "check" method, since it has nothing to do with checking and is a necessary part of constructing the program model.